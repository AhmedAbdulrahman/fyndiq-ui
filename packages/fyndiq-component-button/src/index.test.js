import React from 'react'
import renderer from 'react-test-renderer'

import Button from './'

describe('fyndiq-component-button', () => {
  test('should be rendered without props', () => {
    const component = renderer.create(<Button>Hello</Button>)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should have a type prop', () => {
    const component = renderer.create(<Button type="primary">Hello</Button>)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should have a size prop', () => {
    const component = renderer.create(<Button size="xs">Hello</Button>)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should have a horizontal prop', () => {
    const component = renderer.create(<Button horizontal>Hello</Button>)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should have a disabled prop', () => {
    const component = renderer.create(<Button disabled>Hello</Button>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.props.disabled).toBe(true)
  })

  test('should have a pressed prop', () => {
    const component = renderer.create(<Button pressed>Hello</Button>)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
