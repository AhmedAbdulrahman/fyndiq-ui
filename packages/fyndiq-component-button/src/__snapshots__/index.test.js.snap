// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-button should be rendered without props 1`] = `
<button
  className="
      button
      type-normal
      size-s
      false
      false
      interactive
    "
  disabled={false}
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`fyndiq-component-button should have a disabled prop 1`] = `
<button
  className="
      button
      type-normal
      size-s
      false
      false
      disabled
    "
  disabled={true}
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`fyndiq-component-button should have a horizontal prop 1`] = `
<button
  className="
      button
      type-normal
      size-s
      horizontal
      false
      interactive
    "
  disabled={false}
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`fyndiq-component-button should have a pressed prop 1`] = `
<button
  className="
      button
      type-normal
      size-s
      false
      pressed
      interactive
    "
  disabled={false}
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`fyndiq-component-button should have a size prop 1`] = `
<button
  className="
      button
      type-normal
      size-xs
      false
      false
      interactive
    "
  disabled={false}
  onClick={[Function]}
>
  Hello
</button>
`;

exports[`fyndiq-component-button should have a type prop 1`] = `
<button
  className="
      button
      type-primary
      size-s
      false
      false
      interactive
    "
  disabled={false}
  onClick={[Function]}
>
  Hello
</button>
`;
